ICU library not found when running `cargo pgrx init`

Running `cargo pgrx init` results in an error that seems to indicate a missing ICU library.

I've tried uninstalling `postgresql@15` and `postgresql@16`.

I've tried re-installing icu4c with `brew install icu4c`, setting the following paths and variables:

```
export PATH="/opt/homebrew/opt/icu4c/bin:$PATH"
export PATH="/opt/homebrew/opt/icu4c/sbin:$PATH"
export LDFLAGS="-L/opt/homebrew/opt/icu4c/lib"
export CPPFLAGS="-I/opt/homebrew/opt/icu4c/include"
```

and I've tried force linking it with `brew link --force icu4c`, but I'm still getting the exact same error log below.

## Error log

```
Error:
   0: cd "/Users/bryan/.pgrx/16.1" && CPPFLAGS="-I/opt/homebrew/opt/icu4c/include -DMEMORY_CONTEXT_CHECKING=1 -DCLOBBER_FREED_MEMORY=1 -DRANDOMIZE_ALLOCATED_MEMORY=1 " PATH="/Users/bryan/.pgrx/16.1:/opt/homebrew/opt/icu4c/bin:/opt/homebrew/opt/postgresql@16/bin:/Users/bryan/Applications/miniconda3/condabin:/Users/bryan/.jenv/shims:/Users/bryan/.jenv/bin:/Users/bryan/.zplug/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/bryan/.cargo/bin:/Users/bryan/.rvm/bin:/Users/bryan/.rvm/bin:/Users/bryan/.fzf/bin" PKG_CONFIG_PATH="/opt/homebrew/opt/icu4c/lib/pkgconfig" "/Users/bryan/.pgrx/16.1/configure" "--prefix=/Users/bryan/.pgrx/16.1/pgrx-install" "--with-pgport=28816" "--enable-debug" "--enable-cassert"
      checking build system type... aarch64-apple-darwin23.2.0
      checking host system type... aarch64-apple-darwin23.2.0
      checking which template to use... darwin
      checking whether NLS is wanted... no
      checking for default port number... 28816
      checking for block size... 8kB
      checking for segment size... 1GB
      checking for WAL block size... 8kB
      checking for gcc... gcc
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether gcc accepts -g... yes
      checking for gcc option to accept ISO C89... none needed
      checking for gcc option to accept ISO C99... none needed
      checking for g++... g++
      checking whether we are using the GNU C++ compiler... yes
      checking whether g++ accepts -g... yes
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether gcc supports -Wdeclaration-after-statement, for CFLAGS... yes
      checking whether gcc supports -Werror=vla, for CFLAGS... yes
      checking whether gcc supports -Werror=unguarded-availability-new, for CFLAGS... yes
      checking whether g++ supports -Werror=unguarded-availability-new, for CXXFLAGS... yes
      checking whether gcc supports -Wendif-labels, for CFLAGS... yes
      checking whether g++ supports -Wendif-labels, for CXXFLAGS... yes
      checking whether gcc supports -Wmissing-format-attribute, for CFLAGS... yes
      checking whether g++ supports -Wmissing-format-attribute, for CXXFLAGS... yes
      checking whether gcc supports -Wimplicit-fallthrough=3, for CFLAGS... no
      checking whether g++ supports -Wimplicit-fallthrough=3, for CXXFLAGS... no
      checking whether gcc supports -Wcast-function-type, for CFLAGS... yes
      checking whether g++ supports -Wcast-function-type, for CXXFLAGS... yes
      checking whether gcc supports -Wshadow=compatible-local, for CFLAGS... no
      checking whether g++ supports -Wshadow=compatible-local, for CXXFLAGS... no
      checking whether gcc supports -Wformat-security, for CFLAGS... yes
      checking whether g++ supports -Wformat-security, for CXXFLAGS... yes
      checking whether gcc supports -fno-strict-aliasing, for CFLAGS... yes
      checking whether g++ supports -fno-strict-aliasing, for CXXFLAGS... yes
      checking whether gcc supports -fwrapv, for CFLAGS... yes
      checking whether g++ supports -fwrapv, for CXXFLAGS... yes
      checking whether gcc supports -fexcess-precision=standard, for CFLAGS... no
      checking whether g++ supports -fexcess-precision=standard, for CXXFLAGS... no
      checking whether gcc supports -funroll-loops, for CFLAGS_UNROLL_LOOPS... yes
      checking whether gcc supports -ftree-vectorize, for CFLAGS_VECTORIZE... yes
      checking whether gcc supports -Wunused-command-line-argument, for NOT_THE_CFLAGS... yes
      checking whether gcc supports -Wcompound-token-split-by-macro, for NOT_THE_CFLAGS... yes
      checking whether gcc supports -Wformat-truncation, for NOT_THE_CFLAGS... no
      checking whether gcc supports -Wstringop-truncation, for NOT_THE_CFLAGS... no
      checking whether gcc supports -Wcast-function-type-strict, for NOT_THE_CFLAGS... no
      checking whether gcc supports -fvisibility=hidden, for CFLAGS_SL_MODULE... yes
      checking whether g++ supports -fvisibility=hidden, for CXXFLAGS_SL_MODULE... yes
      checking whether g++ supports -fvisibility-inlines-hidden, for CXXFLAGS_SL_MODULE... yes
      checking whether the C compiler still works... yes
      checking how to run the C preprocessor... gcc -E
      checking for pkg-config... no
      checking allow thread-safe client libraries... yes
      checking whether to build with ICU support... yes
      checking for icu-uc icu-i18n... no
      configure: error: ICU library not found
      If you have ICU already installed, see config.log for details on the
      failure.  It is possible the compiler isn't looking in the proper directory.
      Use --without-icu to disable ICU support.


Location:
   /Users/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.11.2/src/command/init.rs:432

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

   0: cargo_pgrx::command::init::download_postgres with pg_version=16.1
      at /Users/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.11.2/src/command/init.rs:210
   1: cargo_pgrx::command::init::init_pgrx
      at /Users/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.11.2/src/command/init.rs:136
   2: cargo_pgrx::command::init::execute
      at /Users/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.11.2/src/command/init.rs:83

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```


## System information
M2 Macbook Pro
macOS Sonoma 14.2

--- a/README.md
+++ b/README.md
@@ -88,8 +88,8 @@ and `double`. This may not be "unsound" in itself, as it is "merely" illogical,
 but it may undermine otherwise-reasonable safety assumptions of PGRX extensions.
 We do not plan to add support without considerable ongoing technical and financial contributions.
 
-<details>
-   <summary>How to: GCC 7 on CentOS 7</summary>
+<details style="border: 1px solid; padding: 0.25em 0.5em 0;">
+   <summary><i>How to:</i> <b>GCC 7 on CentOS 7</b></summary>
 
 It is not recommended to use CentOS 7 for PGRX development, even if it works.
 
@@ -104,8 +104,8 @@ scl enable devtoolset-7 bash
 ```
 </details>
 
-<details>
-   <summary>How to: Homebrew on macOS</summary>
+<details style="border: 1px solid; padding: 0.25em 0.5em 0;">
+   <summary><i>How to:</i> <b>Homebrew on macOS</b></summary>
 
 As macOS provides no package manager, it is recommended to use https://brew.sh for C dependencies.
 

--- a/README.md
+++ b/README.md
@@ -104,6 +104,18 @@ scl enable devtoolset-7 bash
 ```
 </details>
 
+<details>
+   <summary>How to: Homebrew on macOS</summary>
+
+As macOS provides no package manager, it is recommended to use https://brew.sh for C dependencies.
+
+In particular, you will probably need these if you don't have them already:
+
+```zsh
+brew install git icu4c pkg-config
+```
+</details>
+
 ## Getting Started
 
 

--- a/cargo-pgrx/README.md
+++ b/cargo-pgrx/README.md
@@ -16,6 +16,10 @@ $ cargo install --locked cargo-pgrx
 
 As new versions of `pgrx` are released, you'll want to make sure you run this command again to update it. You should also reinstall `cargo-pgrx` whenever you update `rustc` so that the same compiler is used to build `cargo-pgrx` and your Postgres extensions. You can force `cargo` to reinstall an existing crate by passing `--force`.
 
+Note that some of the features of PGRX involve compiling C code, including `cargo pgrx init`, and
+as such you will also need a toolchain for doing so and potentially must provide various libraries.
+Normally, Rust requires a C toolchain anyways, but it does not require e.g. pkg-config.
+
 ## Usage
 
 ```console

